kong/plugins/prometheus/prometheus.lua

```diff
@@ -493,20 +514,7 @@ function Prometheus:inc(name, label_names, label_values, value)
   local key = full_metric_name(name, label_names, label_values)
   if value == nil then value = 1 end
 
-  local newval, err = self.dict:incr(key, value)
-  if newval then
-    return
-  end
-  -- Yes, this looks like a race, so I guess we might under-report some values
-  -- when multiple workers simultaneously try to create the same metric.
-  -- Hopefully this does not happen too often (shared dictionary does not get
-  -- reset during configuation reload).
-  if err == "not found" then
-    self:set_key(key, value)
-    return
-  end
-  -- Unexpected error
-  self:log_error_kv(key, value, err)
+  self.resty_counter:incr(key, value)
```

Comment:
My understanding is that this is removed because of the per-worker storage of the key,value data and eventual consistency, is that correct?

Reply:
Yes. And plus we don't throw any error from `resty_counter:incr` function.